packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts

```diff
@@ -178,5 +233,164 @@ export function validateMPTokenIssuanceCreate(
       )
     }
   }
+
+  logWarningsForMPTMetadata(tx.MPTokenMetadata)
 }
 /* eslint-enable max-lines-per-function */
+
+/* eslint-disable max-lines-per-function -- Required here as structure validation is verbose. */
+/* eslint-disable max-statements -- Required here as structure validation is verbose. */
+/* eslint-disable no-console -- This function is meant to print console warnings. */
+/**
+ * Logs warning messages if MPTokenMetadata does not adhere to XLS-89d standard.
+ *
+ * @param input - hex encoded MPTokenMetadata.
+ *
+ */
+function logWarningsForMPTMetadata(input?: string): void {
+  if (input == null) {
+    return
+  }
+
+  let jsonMetaData: unknown
+
+  try {
+    jsonMetaData = JSON.parse(hexToString(input))
+  } catch (_err) {
+    console.warn(
+      'MPTokenMetadata is not properly formatted as JSON as per the XLS-89d standard. ',
+    )
+    return
+  }
+
+  if (
+    !(
+      jsonMetaData != null &&
+      typeof jsonMetaData === 'object' &&
+      !Array.isArray(jsonMetaData)
+    )
+  ) {
+    console.warn(
+      'MPTokenMetadata is not properly formatted as JSON as per the XLS-89d standard. ',
```

Comment:
Could you rephrase the warning message?
```suggestion
      'MPTokenMetadata is not properly formatted as JSON as per the XLS-89d standard. While adherence to this standard is not mandatory, such non-compliant MPToken's might not be discoverable by Explorers and Indexers in the XRPL ecosystem.',
```

Reply:
Handled this suggestion as console message header in the updated PR.